src/Components/NavBar/NavItem.tsx

```diff
@@ -64,13 +76,13 @@ export const NavItem: React.FC<NavItemProps> = ({
         </MenuContainer>
       )}
 
-      {showOverlay && <Overlay />}
+      {showOverlay && <Overlay hover={hover} />}
```

Comment:
Also pass UI behavior over to `<NavItem Overlay={NotificationBadge} />` components